src/elfpy/pricing_models/base.py

```diff
@@ -0,0 +1,223 @@
+"""The base pricing model."""
+
+from abc import ABC, abstractmethod
+
+from elfpy.types import Quantity, MarketState, StretchedTime, TradeResult
+import elfpy.utils.price as price_utils
+
+# TODO: Currently many functions use >5 arguments.
```

Comment:
Can you check these pylint disables & TODOS? They all might be resolved. If not, leave them.